File: 3. Array/ArrayReversal.cpp
#include <iostream>
#include <stdio.h>
#include <conio.h>

void swap(int &a, int &b)
{
    int c = a;
    a = b;
    b = c;
}

void ReverseArray(int arr[], int n)
{
    int l=0;
    int r= n-1;

    while(l<r)
    {
        swap(arr[l], arr[r]);
        l++;
        r--;
    }
}

int main()
{
    system("cls");

    int arr[] = {64, 34, 25, 12, 22, 11, 90};
    int n = sizeof(arr) / sizeof(arr[0]);

    ReverseArray(arr, n);

    for (int i = 0; i <= n - 1; i++)
    {
        std::cout << arr[i] << " ";
    }

    return 0;
}

File: 3. Array/ArrayReversal.py
def ReverseArray(lst, n):

    l=0
    r=n-1

    while(l<r):

        lst[l], lst[r] = lst[r], lst[l]
        l+=1
        r-=1

arr = [64, 34, 25, 12, 22, 11, 90]

n = len(arr)

ReverseArray(arr, n)

print(arr)